src/ucp/wireup/wireup.c

```diff
@@ -801,20 +790,29 @@ ucp_wireup_connect_lane_to_iface(ucp_ep_h ep, ucp_lane_index_t lane,
         return status;
     }
 
-    /* If EP already exists, it's a wireup proxy (only client-server),
-     * and we need to update its next_ep instead of replacing it.
-     */
+    if ((ep->uct_eps[lane] == NULL) && ucp_ep_has_cm_lane(ep)) {
+        /* create wireup EP in case of CM lane is used, since a WIREUP EP
+         * is used to keep user's pending requests and send WIREUP MSGs (if
+         * this is WIREUP MSG lane) until CM and WIREUP_MSG phases are done */
+        status = ucp_wireup_ep_create(ep, &ep->uct_eps[lane]);
+        if (status != UCS_OK) {
+            /* coverity[leaked_storage] */
+            return status;
+        }
+    }
+
     if (ep->uct_eps[lane] == NULL) {
         ucs_trace("ep %p: assign uct_ep[%d]=%p", ep, lane, uct_ep);
         ep->uct_eps[lane] = uct_ep;
     } else {
-        ucs_assert(ucp_wireup_ep_test(ep->uct_eps[lane]) &&
-                   (ucp_ep_has_cm_lane(ep) || !ucp_ep_is_sockaddr_stub(ep)));
+        /* If EP already exists, it's a wireup proxy, and we need to update
+         * its next_ep instead of replacing it.
+         */
+        ucs_assert(ucp_wireup_ep_test(ep->uct_eps[lane]));
```

Comment:
can we do like this
ucs_assert(ep->uct_eps[lane] == NULL)
if (ucp_ep_has_cm_lane(ep)) {
   ucp_wireup_ep_create
   ucp_wireup_ep_set_next_ep
   ...
} else {
  ep->uct_eps[lane] = uct_ep;
}